crates/sui-config/src/lib.rs

```diff
@@ -0,0 +1,501 @@
+                    .iter()
+                    .find(|validator| validator.public_key() == *key.public_key_bytes())
+                    .map(|validator| validator.network_address().clone())
+                    .unwrap();
+                let consensus_address = committe_config
+                    .narwhal_committee()
+                    .authorities
+                    .get(&key.public_key_bytes().make_narwhal_public_key().unwrap())
+                    .unwrap()
+                    .workers
+                    .get(&0)
+                    .unwrap()
+                    .transactions
+                    .clone();
+                let consensus_db_path = config_dir
+                    .join(CONSENSUS_DB_NAME)
+                    .join(encode_bytes_hex(key.public_key_bytes()));
+                let consensus_config = ConsensuseConfig {
+                    consensus_address,
+                    consensus_db_path,
+                    narwhal_config: Default::default(),
+                };
+
+                let metrics_address = new_network_address();
+
+                ValidatorConfig {
+                    key_pair: key,
+                    db_path,
+                    network_address,
+                    metrics_address,
+                    consensus_config,
+                    committee_config: committe_config.clone(),
+                }
+            })
+            .collect();
+
+        Self {
+            validator_configs,
+            loaded_move_packages: vec![],
+        }
+    }
+
+    pub fn generate(config_dir: &Path, quorum_size: usize) -> Self {
+        Self::generate_with_rng(config_dir, quorum_size, OsRng)
+    }
+}
+
+// pub struct ConfigBuilder<R> {
+//     rng: R,
+
+// }
+
+fn new_network_address() -> Multiaddr {
+    format!("/dns/localhost/tcp/{}/http", utils::get_available_port())
+        .parse()
+        .unwrap()
+}
+
+const SUI_DIR: &str = ".sui";
+const SUI_CONFIG_DIR: &str = "sui_config";
+pub const SUI_NETWORK_CONFIG: &str = "network.conf";
+pub const SUI_WALLET_CONFIG: &str = "wallet.conf";
+pub const SUI_GATEWAY_CONFIG: &str = "gateway.conf";
+pub const SUI_DEV_NET_URL: &str = "https://gateway.devnet.sui.io:9000";
+
+pub const AUTHORITIES_DB_NAME: &str = "authorities_db";
+pub const DEFAULT_STARTING_PORT: u16 = 10000;
+pub const CONSENSUS_DB_NAME: &str = "consensus_db";
+
+pub fn sui_config_dir() -> Result<PathBuf, anyhow::Error> {
+    match std::env::var_os("SUI_CONFIG_DIR") {
+        Some(config_env) => Ok(config_env.into()),
+        None => match dirs::home_dir() {
+            Some(v) => Ok(v.join(SUI_DIR).join(SUI_CONFIG_DIR)),
+            None => anyhow::bail!("Cannot obtain home directory path"),
+        },
+    }
+    .and_then(|dir| {
+        if !dir.exists() {
+            std::fs::create_dir_all(dir.clone())?;
+        }
+        Ok(dir)
+    })
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct GenesisConfig {
+    pub quorum_size: usize,
```

Comment:
`committee_size` is probably better - `quorum_size` sounds like the `k` in `k out of n`